content/unified_mode.md

```diff
@@ -0,0 +1,363 @@
++++
+title = "Unified Mode"
+draft = false
+
+aliases = ["/unified_mode.html"]
+
+[menu]
+  [menu.infra]
+    title = "Unified Mode Guide"
+    identifier = "chef_infra/cookbook_reference/resources/unified_mode.md Unified Modee"
+    parent = "chef_infra/cookbook_reference/resources"
+    weight = 50
++++
+
+[\[edit on GitHub\]](https://github.com/chef/chef-web-docs/blob/master/content/unified_mode.md)
+
+## Unified Mode
+
+Unified mode is a setting which eliminates the distinction between the compile and converge modes
+in the way that Chef Infra recipe and resources are parsed.  The setting replaces Chef Infra's two pass
+parsing with single pass parsing, so that resources are executed as soon as they are declared.
+
+This results in considerably clearer code, with considerably less Ruby knowledge required to understand
+the order of operations.
+
+### Unified Mode for Custom Resources
+
+Unified mode can be turned on for a resource, with the `unified_mode` method, which takes a Boolean argument:
+
+```ruby
+unifed_mode true
+
+provides :my_resource_name
+
+action :run do
+  # implementation code goes here
+end
+```
+
+### Unified Mode for Recipe Code
+
+This has not yet been written, so there is no setting for this yet.  It is planned to be delivered in 2020.
+Currently, `unified_mode` can only be set for resources.
+
+### Unified Mode Isolation
+
+If a `unified_mode` resource calls a non-`unified_mode` resource the called resource is not executed in
+`unified_mode`.  Each resource maintains its own state as being in `unified_mode` or not.  Custom
+Resources do not need to worry about if their calling state is in `unified_mode` or not since that
+calling context will not affect the resources execution.  Resources using `unified_mode` may call
+resources not using `unified_mode` and vice versa.
+
+## Benefits of Unified Mode
+
+### Single Pass Execution
+
+In Unified Mode the Chef Infra Language executes from top to bottom, eliminating the compile and converge phases.
+
+With the deferred execution of resources to converge time, the user has to understand many different details
+of the Ruby parser to understand what constructs relate to Chef Infra Resources and what constructs are parts of
+the core Ruby language to determine when those expression are executed.  All of that complexity is removed
+in Unified Mode.
+
+### Elimination of Lazy Constructs
+
+Several aspects of the Chef Infra Language still work, but are no longer useful in Unified Mode.  The need for
+`lazy` blocks is entirely eliminated.  The need to lazy ruby code via a `ruby_block` is also eliminated.
```

Comment:
I don't actually think that's true. I've seen it still required in a few places.

[Referencing properties in other properties still seems to require lazy eval](https://github.com/sous-chefs/haproxy/pull/457/files/7d0f00881c3c0559345c6bd0159fa4ff2d42554d)